sdk/eventhub/Azure.Messaging.EventHubs.Experimental/src/Producer/IdempotentProducer.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Producer;
+
+namespace Azure.Messaging.EventHubs.Experimental
+{
+    /// <summary>
+    ///   A client responsible for publishing <see cref="EventData" /> to a specific Event Hub,
+    ///   grouped together in batches.  Depending on the options specified when sending, events may
+    ///   be automatically assigned an available partition or may request a specific partition.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   <list type="bullet">
+    ///     <listheader><description>Allowing automatic routing of partitions is recommended when:</description></listheader>
+    ///     <item><description>The sending of events needs to be highly available.</description></item>
+    ///     <item><description>The event data should be evenly distributed among all available partitions.</description></item>
+    ///   </list>
+    ///
+    ///   <list type="number">
+    ///     <listheader><description>If no partition is specified, the following rules are used for automatically selecting one:</description></listheader>
+    ///     <item><description>Distribute the events equally amongst all available partitions using a round-robin approach.</description></item>
+    ///     <item><description>If a partition becomes unavailable, the Event Hubs service will automatically detect it and forward the message to another available partition.</description></item>
+    ///   </list>
+    ///
+    ///   <para>
+    ///     The <see cref="IdempotentProducer" /> is safe to cache and use for the lifetime of an application, and that is best practice when the application
+    ///     publishes events regularly or semi-regularly.  The producer holds responsibility for efficient resource management, working to keep resource usage low during
+    ///     periods of inactivity and manage health during periods of higher use.  Calling either the <see cref="EventHubProducerClient.CloseAsync" /> or <see cref="EventHubProducerClient.DisposeAsync" />
+    ///     method as the application is shutting down will ensure that network resources and other unmanaged objects are properly cleaned up.
+    ///   </para>
+    /// </remarks>
+    ///
+    public class IdempotentProducer : EventHubProducerClient
```

Comment:
We didn't explicitly discuss naming;  I went with something that I felt was obvious for the intended partner audience, following the `PartitionReceiver` discussion in the choice to leave off the `Client` suffix here.